### PR TITLE
Refactored modules to hooks

### LINKED ISSUE #98: [FEATURE] Remove "Uwuified text:" in ~uwuify
**Is your feature request related to a problem? Please describe.**
None

**Describe the solution you'd like**
~uwuify to not say "**Uwuified text:**" before my text.

**Describe alternatives you've considered**
None

**Additional context**
Suggested by @Shadowplays4k

### LINKED ISSUE #107: refactor: rename "modules" to "hooks"
[not following issue template as this is different then the usual]

`modules` is confusing, maybe we name them `hooks` and add more functionality to them like attaching to events other then `message`?

### PR DESCRIPTION
**Pull request by LostNuke**
**Does this Pull Request fix an issue? If so, please mention the issue.**
Closes #107
Closes #98 

**Describe your changes.**
I refactored Modules to hooks. All that I found, leaving those Modules that were what I suppose meant to be like that (NPM Modules)

**Describe alternatives you've considered**
I used ctrl+f to look for the modules (a lot of code ye know xD)

### CODE REVIEW

**Comment by thetayloredman** [comment]
Well, you need to rename `src/modules/` to `src/hooks/` and rename `templateModule.ts`

**Review by thetayloredman — CHANGES_REQUESTED** [review]
You need to rename `src/modules/` to `src/hooks/` and rename `templateModule.ts`

**Comment by LostNuke** [comment]
Whoops didnt see those. Also I accidentally already pushed the uwuify thing forgot it would go into here too xD But yeah issue #98 would be solved too

**Comment by thetayloredman** [comment]
Fixes #107
Fixes #98

**Comment by thetayloredman** [comment]
This *should* work, but I need to test.

**Comment by thetayloredman** [comment]
`src/lib/types/augments.ts` needs to be updated as well

**Comment by thetayloredman** [comment]
Here's a `git diff` you can apply:

```diff
diff --git a/src/lib/interfaces/Module.ts b/src/lib/interfaces/Module.ts
deleted file mode 100644
index 0b29622..0000000
--- a/src/lib/interfaces/Module.ts
+++ /dev/null
@@ -1,24 +0,0 @@
-/**
- * ProtoBot -- A Discord furry bot
- * Copyright (C) 2020  BadBoyHaloCat
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <https://www.gnu.org/licenses/>.
- */
-
-import type { Client, Message } from 'discord.js';
-
-export default interface Module {
-    config: { name: string; description: string };
-    run: (client: Client, message: Message, log: (mode: 'i' | 'w' | 'e', message: string) => void) => void;
-}
diff --git a/src/lib/types/Augments.d.ts b/src/lib/types/Augments.d.ts
index 63abff6..6baa796 100644
--- a/src/lib/types/Augments.d.ts
+++ b/src/lib/types/Augments.d.ts
@@ -24,7 +24,7 @@ import Cooldowns from '@lib/interfaces/Cooldowns';
 import Enmap from 'enmap';
 import Fursona from '@lib/interfaces/Fursona';
 import MarkovData from '@lib/interfaces/MarkovData';
-import Module from '@lib/interfaces/Module';
+import Hook from '@lib/interfaces/Hook';
 import UserConfig from '@lib/interfaces/UserConfig';
 import UserStats from '@lib/interfaces/UserStats';
 
@@ -50,7 +50,7 @@ declare module 'discord.js' {
         commands: Enmap<string, Command>;
         commandsConfig: Enmap<string, CommandConfig>; // NOTE: is this redundant?
         commandsRefs: Enmap<string, string>;
-        modules: Enmap<string, Module>;
+        hooks: Enmap<string, Hook>;
 
         // Functions
         public closeDatabases(): void;

```

Put this in a file and then run `git apply path/to/diff.diff` and it'll apply my changes

**Comment by LostNuke** [comment]
OH right that one confused me speaking of uhh.
grafik

**Comment by thetayloredman** [comment]
> OH right that one confused me speaking of uhh.
> grafik

Apply my patch (add a newline after it if you get corrupt patch at line 26) and it'll fix that

**Comment by LostNuke** [comment]
Patch has been applied

**Comment by thetayloredman** [comment]
@Alcremie please take a look at this ok thx

File: src/index.ts [hunk 1]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
             log(type, `${chalk.yellow('[')}${chalk.yellow.bold('MODLOAD')}${chalk.yellow(']')} ${message}`);
```
Comment (thetayloredman) [inline src/index.ts:138]:
```suggestion
            log(type, `${chalk.yellow('[')}${chalk.yellow.bold('HOOKLOAD'}${chalk.yellow(']')} ${message}`);
```

`MODLOAD` is "Module Load", not "Hook Load".

File: src/index.ts [hunk 2]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
...
         }
         l('i', 'Beginning initial module load...');
```
Comment (thetayloredman) [inline src/index.ts:140]:
```suggestion
        l('i', 'Beginning initial hook load...');
```

File: src/index.ts [hunk 3]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
...
                     if (path.endsWith('.js')) {
                         // normal load
-                        const moduleData = require(client.config.dirs.modules.endsWith('/')
-                            ? client.config.dirs.modules + path
-                            : `${client.config.dirs.modules}/${path}`);
+                        const moduleData = require(client.config.dirs.hooks.endsWith('/')
+                            ? client.config.dirs.hooks + path
+                            : `${client.config.dirs.hooks}/${path}`);
                         const modName = path.replace('.js', '');
                         l('i', `Loading module "${modName}"...`);
```
Comment (thetayloredman) [inline src/index.ts:153]:
```suggestion
                        l('i', `Loading hook "${modName}"...`);
```

File: src/index.ts [hunk 4]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
...
+                l('e', `Failed to read directory ${client.config.dirs.hooks}:`);
                 l('e', err);
             } else {
                 files.forEach((path: string) => {
                     if (path.endsWith('.js')) {
                         // normal load
-                        const moduleData = require(client.config.dirs.modules.endsWith('/')
-                            ? client.config.dirs.modules + path
-                            : `${client.config.dirs.modules}/${path}`);
+                        const moduleData = require(client.config.dirs.hooks.endsWith('/')
```
Comment (thetayloredman) [inline src/index.ts:149]:
```suggestion
                        const hookData = require(client.config.dirs.hooks.endsWith('/')
```

File: src/index.ts [hunk 5]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
...
+                            ? client.config.dirs.hooks + path
+                            : `${client.config.dirs.hooks}/${path}`);
                         const modName = path.replace('.js', '');
```
Comment (thetayloredman) [inline src/index.ts:152]:
```suggestion
                        const hookName = path.replace('.js', '');
```

File: src/index.ts [hunk 6]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
...
                         l('i', `Loading module "${modName}"...`);
-                        client.modules.set(modName, moduleData);
+                        client.hooks.set(modName, moduleData);
```
Comment (thetayloredman) [inline src/index.ts:154]:
```suggestion
                        client.hooks.set(hookName, hookData);
```

File: src/index.ts [hunk 7]
```diff
@@ -138,26 +138,26 @@ client.on('ready', async () => {
...
                         l('i', `Finished loading module "${modName}"!`);
```
Comment (thetayloredman) [inline src/index.ts:155]:
```suggestion
                        l('i', `Finished loading hook "${hookName}"!`);
```

File: src/index.ts [hunk 8]
```diff
@@ -179,7 +179,7 @@ client.on('message', (message: discord.Message) => {
         return;
     }
     // @ts-ignore
-    client.modules.forEach((moduleData) => {
+    client.hooks.forEach((moduleData) => {
         log('i', `${chalk.green('[')}${chalk.green.bold('ModuleRunner')}${chalk.green(']')} Running module ${moduleData.config.name}!`);
```
Comment (thetayloredman) [inline src/index.ts:183]:
```suggestion
        log('i', `${chalk.green('[')}${chalk.green.bold('HookRunner')}${chalk.green(']')} Running hook ${hookData.config.name}!`);
```

File: src/index.ts [hunk 9]
```diff
@@ -179,7 +179,7 @@ client.on('message', (message: discord.Message) => {
         return;
     }
     // @ts-ignore
-    client.modules.forEach((moduleData) => {
+    client.hooks.forEach((moduleData) => {
```
Comment (thetayloredman) [inline src/index.ts:182]:
```suggestion
    client.hooks.forEach((hookData) => {
```

File: src/index.ts [hunk 10]
```diff
@@ -179,7 +179,7 @@ client.on('message', (message: discord.Message) => {
...
         log('i', `${chalk.green('[')}${chalk.green.bold('ModuleRunner')}${chalk.green(']')} Running module ${moduleData.config.name}!`);
         moduleData.run(client, message, log);
```
Comment (thetayloredman) [inline src/index.ts:184]:
```suggestion
        hookData.run(client, message, log);
```

**Review by thetayloredman — CHANGES_REQUESTED** [review]
A few more things need to be addressed.

**Comment by thetayloredman** [comment]
You can batch them you know...

File: src/index.ts [hunk 11]
```diff
@@ -135,29 +135,29 @@ client.on('ready', async () => {
     loadCmds();
     function loadMods(): void {
```
Comment (thetayloredman) [inline src/index.ts:136]:
```suggestion
    function loadHooks(): void {
```

Also change line 166 manually (I can't)

**Review by thetayloredman — CHANGES_REQUESTED** [review]
Function name too!

**Review by thetayloredman — APPROVED** [review]
LGTM! @Alcremie please review

File: src/commands/uwuify.ts [hunk 12]
```diff
@@ -29,7 +29,7 @@ export function run(client: Client, message: Message, args: string[], log: (mode
     const uwuify: Uwuifier = new Uwuifier();
     const msg: string = uwuify.uwuifySentence(args.join(' '));
     // @ts-ignore
-    message.channel.send(`**Uwuified text:** ${msg}`);
+    message.channel.send(`${msg}`);
```
Comment (Alcremie) [inline src/commands/uwuify.ts:32]: Interpolating `msg` is redundant; it's already a string.
```diff
- message.channel.send(`${msg}`);
+ message.channel.send(msg);
```
There's also really no need to explicitly type either `uwuify` or `msg` in the former lines, but that's beyond the scope of the PR.

File: src/index.ts [hunk 13]
```diff
@@ -133,37 +133,37 @@ client.on('ready', async () => {
         });
     }
     loadCmds();
-    function loadMods(): void {
+    function loadHooks(): void {
         function l(type: 'i' | 'w' | 'e', message: any) {
-            log(type, `${chalk.yellow('[')}${chalk.yellow.bold('MODLOAD')}${chalk.yellow(']')} ${message}`);
+            log(type, `${chalk.yellow('[')}${chalk.yellow.bold('HOOKLOAD'}${chalk.yellow(']')} ${message}`);
```
Comment (Alcremie) [inline src/index.ts]: You missed a closing parentheses after `'HOOKLOAD'`.

Comment (thetayloredman) [inline src/index.ts:138]:
```suggestion
            log(type, `${chalk.yellow('[')}${chalk.yellow.bold('HOOKLOAD')}${chalk.yellow(']')} ${message}`);
```

Syntax error

**Review by thetayloredman — CHANGES_REQUESTED** [review]

**Review by Alcremie — COMMENTED** [review]
A few extra things that needed to be considered.

File: src/index.ts (hunk 13 above)
Comment (thetayloredman) [reply src/index.ts]: Already pushed that as a hotfix lol

File: src/index.ts [hunk 14]
```diff
@@ -133,37 +133,37 @@ client.on('ready', async () => {
...
-                            : `${client.config.dirs.modules}/${path}`);
-                        const modName = path.replace('.js', '');
-                        l('i', `Loading module "${modName}"...`);
-                        client.modules.set(modName, moduleData);
-                        l('i', `Finished loading module "${modName}"!`);
+                        const hookData = require(client.config.dirs.hooks.endsWith('/')
+                            ? client.config.dirs.hooks + path
+                            : `${client.config.dirs.hooks}/${path}`);
+                        const hookName = path.replace('.js', '');
+                        l('i', `Loading hook "${modName}"...`);
```
Comment (thetayloredman) [inline src/index.ts:153]:
```suggestion
                        l('i', `Loading hook "${hookName}"...`);
```

**Review by thetayloredman — CHANGES_REQUESTED** [review]
My brain is very large

**Review by thetayloredman — APPROVED** [review]
Tested it and it appears to work! :clap:

**Comment by thetayloredman** [comment]
@all-contributors please add @LostNuke for code

**Comment by thetayloredman** [comment]
@all-contributors please add @Alcremie for maintenance and code

**Pull request merged**